assets/js/core/WakaTimeCore.js

```diff
@@ -83,32 +77,35 @@ class WakaTimeCore {
      * and sends it to WakaTime for logging.
      */
     recordHeartbeat() {
-
         chrome.storage.sync.get({
             loggingEnabled: config.loggingEnabled,
             loggingStyle: config.loggingStyle,
             blacklist: '',
             whitelist: ''
         }, (items) => {
             if (items.loggingEnabled === true) {
-
                 changeExtensionState('allGood');
 
                 chrome.idle.queryState(config.detectionIntervalInSeconds, (newState) => {
-
                     if (newState === 'active') {
                         // Get current tab URL.
                         chrome.tabs.query({active: true}, (tabs) => {
 
                             var currentActiveTab = tabs[0];
+                            var debug = false, entity;
 
-                            var debug = false;
                             // If the current active tab has devtools open
-                            if (in_array(currentActiveTab.id, this.tabsWithDevtoolsOpen)) debug = true;
+                            if (in_array(currentActiveTab.id, this.tabsWithDevtoolsOpen)) {
+                                debug = true;
+                            }
 
                             if (items.loggingStyle == 'blacklist') {
-                                if (! contains(currentActiveTab.url, items.blacklist)) {
-                                    this.sendHeartbeat(currentActiveTab.url, debug);
+                                entity = this.checkURL(currentActiveTab.url, items.blacklist);
```

Comment:
Why does the blacklist also need to go through `checkURL`? Could we leave it using `contains`?